### PR TITLE
i.MX Platform code cleanup

### PR DESCRIPTION
**Pull request by clementfaure**
Hello,
In this pull-request, I remove the power management code for imx7d platform. It is outdated, unmaintained and surely non-working. If someday, I upstream the more recent power management functionalities, that will be done in `core/drivers/pm`.

I also move some drivers from `plat-imx` to `drivers`. I removed the misleading `drivers` directory in `plat-imx` as all drivers should go to `core/drivers` (if possible). 

Thanks

### CODE REVIEW

File: core/arch/arm/plat-imx/imx_sip.h [hunk 1]
```diff
@@ -6,10 +6,4 @@
 #ifndef __IMX_SIP_H__
 #define __IMX_SIP_H__
 
-#define IMX_SIP_PL310_ENABLE			1
-#define IMX_SIP_PL310_DISABLE			2
-#define IMX_SIP_PL310_ENABLE_WRITEBACK		3
-#define IMX_SIP_PL310_DISABLE_WRITEBACK		4
-#define IMX_SIP_PL310_ENABLE_WFLZ		5
-
```
Comment (jforissier) [inline core/arch/arm/plat-imx/imx_sip.h:14]: Should go in the next commit

**Review by jforissier — COMMENTED** [review]
Reviewing "core: imx: remove power management code for imx7 platforms"

It looks like this removes much more than i.MX7 code only (`pcsi.c` has `#ifdef CFG_IMX& ... #else ... #endif` for instance). Plus one comment below.

File: core/arch/arm/plat-imx/drivers/imx_scu.c [hunk 2]
```diff
@@ -24,9 +30,8 @@ static TEE_Result scu_init(void)
 	io_write32(scu_base + SCU_NSAC, SCU_NSAC_CTRL_INIT);
 
 	/* SCU enable */
-	io_write32(scu_base + SCU_CTRL, io_read32(scu_base + SCU_CTRL) | 0x1);
+	io_setbits32(scu_base + SCU_CTRL, BIT32(0))
```
Comment (jforissier) [inline core/arch/arm/plat-imx/drivers/imx_scu.c]: Unrelated change, should be a separate patch.

**Review by jforissier — COMMENTED** [review]
Reviewing "core: imx: scu: move SCU configuration values to source file".

With the below comment addressed:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by jforissier — COMMENTED** [review]
For commit "core: imx: caam: remove imx_caam.h header file":
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by jforissier — COMMENTED** [review]
For commit "driver: imx_caam: move imx_caam driver source file":
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by jforissier — COMMENTED** [review]
For commit "drivers: imx_scu: move imx_scu driver source file" with the below comment addressed:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by jforissier — COMMENTED** [review]
For commit "drivers: imx_csu: move imx_csu driver source file":
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by jforissier — COMMENTED** [review]
For commit "core: imx: remove drivers directory":
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/arch/arm/plat-imx/imx_sip.h (hunk 1 above)
Comment (clementfaure) [reply core/arch/arm/plat-imx/imx_sip.h:14]: Correct

File: core/arch/arm/plat-imx/drivers/imx_scu.c (hunk 2 above)
Comment (clementfaure) [reply core/arch/arm/plat-imx/drivers/imx_scu.c]: I will remove that chage

File: core/drivers/imx_scu.c [hunk 3]
```diff
@@ -30,7 +30,7 @@ static TEE_Result scu_init(void)
 	io_write32(scu_base + SCU_NSAC, SCU_NSAC_CTRL_INIT);
 
 	/* SCU enable */
-	io_setbits32(scu_base + SCU_CTRL, BIT32(0))
+	io_setbits32(scu_base + SCU_CTRL, BIT32(0));
```
[the remark this answers is not in the document]
Comment (clementfaure) [reply core/drivers/imx_scu.c:33]: I will remove it

**Comment by clementfaure** [comment]
Tags applied + rebase
